playbooks/openstack/openshift-cluster/provision.yml

```diff
@@ -47,6 +50,13 @@
   hosts: oo_all_hosts
   become: yes
   gather_facts: yes
+  roles:
+  - role: rhel_subscribe
+    when:
+    - ansible_distribution == "RedHat"
+    - lookup('env', 'rhel_subscription_user') | default(rhsub_user | default(False, True), True)
```

Comment:
could you please elaborate why these have to be consumed as ENV vars instead of ansible inventory vars? Do we need a docs update as well?

Reply:
Because that's what the role does. Take a look at the roles/rhel_subscription/tasks/main.yml file. If we didn't read the environment variable here as well, the behaviour would be inconsistent.

I would prefer if the role just used the vars and defaults as usual, but that would be a breaking change and should be done in a different PR (assuming the project wants to do that).

Reply:
yeah, let's get a PR filed to remove reading from the environment automatically.  If a users wishes to do this, they can add lookup() to the variable itself.